pkg/controller/rollout/rolling.go

```diff
@@ -0,0 +1,162 @@
+package rollout
+
+import (
+	"time"
+
+	rolloutv1alpha1 "github.com/openkruise/rollouts/api/v1alpha1"
+	"github.com/openkruise/rollouts/pkg/controller/rollout/batchrelease"
+	"github.com/openkruise/rollouts/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (r *RolloutReconciler) doProgressingInRolling(rollout *rolloutv1alpha1.Rollout, workload *util.Workload, newStatus *rolloutv1alpha1.RolloutStatus) (*time.Time, error) {
+	// Handle the 5 special cases firstly, and we had better keep the order of following cases:
+
+	switch {
+	// 1. In case of rollback in a quick way, un-paused and just use workload rolling strategy
+	case isRollingBackDirectly(rollout, workload):
+		return r.handleRollbackDirectly(rollout, workload, newStatus)
+
+	// 2. In case of rollback in a batch way, use rollout step strategy
+	case isRollingBackInBatches(rollout, workload):
```

Comment:
Does the canaryRevision value need to be adjusted in this scenario? I think there is still a problem if traffic scheduling is needed in this way. Are stable and canary all the same?